src/measure_theory/simple_func_dense.lean

```diff
@@ -667,6 +676,58 @@ by simpa [to_Lp_to_simple_func] using norm_to_Lp (to_simple_func f) (simple_func
 
 end to_simple_func
 
+section induction
+
+variables (p)
+
+/-- The characteristic function of a finite-measure measurable set `s`, as an `Lp` simple function.
+-/
+def indicator_const {s : set α} (hs : measurable_set s) (hμs : μ s ≠ ∞) (c : E) :
+  Lp.simple_func E p μ :=
+to_Lp ((simple_func.const _ c).piecewise s hs (simple_func.const _ 0))
+  (mem_ℒp_indicator_const p hs c (or.inr hμs))
+
+variables {p}
+
+@[simp] lemma coe_indicator_const {s : set α} (hs : measurable_set s) (hμs : μ s ≠ ∞) (c : E) :
+  (↑(indicator_const p hs hμs c) : Lp E p μ) = indicator_const_Lp p hs hμs c :=
+rfl
+
+lemma to_simple_func_indicator_const {s : set α} (hs : measurable_set s) (hμs : μ s ≠ ∞) (c : E) :
+  to_simple_func (indicator_const p hs hμs c)
+    =ᵐ[μ] (simple_func.const _ c).piecewise s hs (simple_func.const _ 0) :=
+Lp.simple_func.to_simple_func_to_Lp _ _
+
+/-- To prove something for an arbitrary `Lp` simple function, with `0 < p < ∞`, it suffices to show
+that the property holds for (multiples of) characteristic functions of finite-measure measurable
+sets and is closed under addition (of functions with disjoint support). -/
+@[elab_as_eliminator]
+protected lemma induction (hp_pos : 0 < p) (hp_ne_top : p ≠ ∞) {P : Lp.simple_func E p μ → Prop}
+  (h_ind : ∀ (c : E) {s : set α} (hs : measurable_set s) (hμs : μ s < ∞),
```

Comment:
There was a discussion on Zulip that a good convention is to have assumptions of theorems in the form `μ s ≠ ∞`, and conclusions in the form `μ s < ∞` (because it's very easy to go from the latter from the former, and slightly more painful in the other direction). So here it would be `(hs : μ s ≠ ∞)`. Same thing in the next lemma.

Reply:
I agree with the principle, but in this case, it's a hypothesis within a hypothesis, which I argue should be treated like a goal!  (i.e., with `μ s < ∞`).  That is, we want `induction` to have the weakest possible hypotheses, so `h_ind` should be weak, so the hypotheses of `h_ind` should be strong.  Do you agree?

Reply:
Yes, very good point!